CIScripts/Common/Invoke-CommandWithFunctions.Tests.ps1

```diff
@@ -0,0 +1,126 @@
+Param (
+    [Parameter(Mandatory=$false)] [string] $TestenvConfFile,
+    [Parameter(ValueFromRemainingArguments=$true)] $UnusedParams
+)
+
+. $PSScriptRoot/Init.ps1
+. $PSScriptRoot/../Testenv/Testenv.ps1
+. $PSScriptRoot/../Testenv/Testbed.ps1
+
+. $PSScriptRoot/Invoke-CommandWithFunctions.ps1
+
+Describe "Invoke-CommandWithFunctions tests" -Tags CI, Systest {
+    function Test-CWF {
+        param(
+            [Parameter(Mandatory=$true)] [string[]] $Functions,
+            [Parameter(Mandatory=$true)] [ScriptBlock] $ScriptBlock,
+            [Switch] $CaptureOutput
+        )
+
+        Invoke-CommandWithFunctions -Session $Session `
+            -FunctionNames $Functions `
+            -ScriptBlock $ScriptBlock `
+            -CaptureOutput:$CaptureOutput
+    }
+
+    function Test-SimpleFunction {
+        param(
+            [Parameter(Mandatory=$false)] [string] $SimpleParam = "A simple string to return"
+        )
+
+        return $SimpleParam
+    }
+
+    $TestSimpleFunction = @("Test-SimpleFunction")
+
+    Context "Incorrect function usage handling" {
+        It "throws on nonexisting function" {
+            { Test-CWF -Functions $TestSimpleFunction  `
```

Comment:
Why not just paste "Test-SimpleFunction" instead of this $TestSimpleFunction variable?

BTW, Powershell will act OK, if you pass a single variable when a function is expecting an array.